src/main/java/com/lothrazar/cyclic/item/ElevationWandItem.java

```diff
@@ -0,0 +1,54 @@
+package com.lothrazar.cyclic.item;
+
+import com.lothrazar.cyclic.ModCyclic;
+import com.lothrazar.cyclic.base.ItemBase;
+import com.lothrazar.cyclic.util.UtilEntity;
+import com.lothrazar.cyclic.util.UtilSound;
+import com.lothrazar.cyclic.util.UtilWorld;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemUseContext;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Hand;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+public class ElevationWandItem extends ItemBase {
+    public ElevationWandItem(Properties properties) {
+        super(properties);
+    }
+
+    @Override
+    @Nonnull
+    public ActionResultType itemInteractionForEntity(ItemStack stack, PlayerEntity playerIn, LivingEntity target, Hand hand) {
+        return tryTeleport(playerIn.world, playerIn, target, stack) ? ActionResultType.SUCCESS : ActionResultType.CONSUME;
+    }
+
+    @Override
+    @Nonnull
+    public ActionResult<ItemStack> onItemRightClick(World worldIn, PlayerEntity playerIn, Hand handIn) {
+        return tryTeleport(worldIn, playerIn, playerIn, playerIn.getHeldItem(handIn)) ?
+                ActionResult.resultSuccess(playerIn.getHeldItem(handIn)) : ActionResult.resultConsume(playerIn.getHeldItem(handIn));
+    }
+
+    private boolean tryTeleport(World world, PlayerEntity playerIn, LivingEntity target, ItemStack stack) {
+        if (target == null || stack == null || world.isRemote)
+            return false;
+        BlockPos destination = UtilWorld.getFirstBlockAbove(world, target.getPosition());
+        if (destination != null) {
+                UtilSound.playSound(target, target.getPosition(), SoundEvents.ITEM_CHORUS_FRUIT_TELEPORT);
```

Comment:
its playing the sound where the player used to be and not destination. might be why the player doesnt hear it

Reply:
even when the player doesn't teleport (because it can't find the destination), it doesn't play the sound. The player hasn't moved in that case.